compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -682,16 +682,304 @@ abstract class GenJSCode extends plugins.PluginComponent
+          case ((numBounds, js.Skip()), acc) => acc
+
+          case ((numBounds, body), acc) =>
+            val cond = mkOverrideNumsCond(overrideNumRef, numBounds)
+            js.If(cond, body, acc)(jstpe.BooleanType)
+        }
+      }
+
+      private[ConstructorGraph] def mkPostSuperCallOnSndCtr(overrideNumRef: js.VarRef,
+          isConstructor: Set[String])(implicit pos: Position): js.Tree = {
+        val postSuperCall = {
+          method.body match {
+            case js.Block(stats) =>
+              stats.dropWhile {
+                case js.ApplyStatic(_, mtd, _) if isConstructor(mtd.name) => false
+                case _                                                    => true
+              }.tail
+
+            case _ => Nil
+          }
+        }
+        js.Block(postSuperCall :+ mkSubPostCalls(overrideNumRef, isConstructor))
+      }
+
+      private def mkOverrideNumsCond(numRef: js.VarRef,
+          numBounds: (Int, Int)) = numBounds match {
+        case (lo, hi) if lo == hi =>
+          js.BinaryOp(js.BinaryOp.===, js.IntLiteral(lo), numRef)
+
+        case (lo, hi) if lo == hi - 1 =>
+          val lhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(lo))
+          val rhs = js.BinaryOp(js.BinaryOp.===, numRef, js.IntLiteral(hi))
+          js.If(lhs, js.BooleanLiteral(true), rhs)(jstpe.BooleanType)
+
+        case (lo, hi) =>
+          val lhs = js.BinaryOp(js.BinaryOp.Num_<=, js.IntLiteral(lo), numRef)
+          val rhs = js.BinaryOp(js.BinaryOp.Num_<=, numRef, js.IntLiteral(hi))
+          js.BinaryOp(js.BinaryOp.Boolean_&, lhs, rhs)
+          js.If(lhs, rhs, js.BooleanLiteral(false))(jstpe.BooleanType)
+      }
+    }
+
+    private def zipMap[T, U, V](xs: List[T], ys: List[U])(
+        f: (T, U) => V): List[V] = {
+      if (xs.nonEmpty && ys.nonEmpty)
+        f(xs.head, ys.head) :: zipMap(xs.tail, ys.tail)(f)
+      else
+        Nil
     }
 
+    private def mkOverloadSelection(constructorGraph: ConstructorGraph,
+        overloadIdent: js.Ident, dispatchResolution: js.Tree,
+        isConstructor: Set[String])(implicit pos: Position): List[js.Tree] = {
+      if (!constructorGraph.hasSubConstructors) {
+        dispatchResolution match {
+          case js.Block(stats) =>
+            val js.ApplyStatic(_, method, _) = stats.last
+            val refs = constructorGraph.getParamRefs(method.name)
+            val rhss = stats.collect { case js.VarDef(_, _, _, rhs) => rhs }
+            zipMap(refs, rhss) { (ref, rhs) =>
+              js.VarDef(ref.ident, ref.tpe, false, rhs)
+            }
+
+          case js.ApplyStatic(cls, method, args) if isConstructor(method.name) =>
+            // Calls the constructor with no arguments
+            Nil
+
+          case tree => List(tree)
+        }
+      } else {
+        val overloadRef = js.VarRef(overloadIdent)(jstpe.IntType)
+        def assignOverloadNumbers(tree: js.Tree): js.Tree = tree match {
```

Comment:
Also, add a blank line before this line (usually there should be a blank line before a local `def`, unless it comes right after an opening `{` of course.